server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/GlobalOrdinalsStringTermsAggregator.java

```diff
@@ -700,11 +700,10 @@ abstract class ResultStrategy<
         private InternalAggregation[] buildAggregations(LongArray owningBucketOrds) throws IOException {
 
             if (valueCount == 0) { // no context in this reader
-                InternalAggregation[] results = new InternalAggregation[Math.toIntExact(owningBucketOrds.size())];
-                for (int ordIdx = 0; ordIdx < results.length; ordIdx++) {
-                    results[ordIdx] = buildNoValuesResult(owningBucketOrds.get(ordIdx));
-                }
-                return results;
+                return GlobalOrdinalsStringTermsAggregator.this.buildAggregations(
```

Comment:
Half of the aggregators aren't using the `Class.this` syntax. I would probably remove it everywhere, as the methods have different params anyway (?). Or rename one of the methods, not sure if it's worth it anyway